src/core/server/logging/logging_config.test.ts

```diff
@@ -171,3 +171,127 @@ test('fails if loggers use unknown appenders.', () => {
 
   expect(() => new LoggingConfig(validateConfig)).toThrowErrorMatchingSnapshot();
 });
+
+describe('extend', () => {
+  it('adds new appenders', () => {
+    const configValue = new LoggingConfig(
+      config.schema.validate({
```

Comment:
Are you referring to the usage of the config schema here? I'm using that mostly to make constructing these configs much simpler. For example:

```ts
    const configValue = new LoggingConfig({
      appenders: new Map([
        [
          'file1',
          {
            kind: 'file',
            layout: { kind: 'pattern' },
            path: 'path',
          },
        ],
      ]),
      loggers: [],
      root: { appenders: [], level: 'debug' },
    });
```

vs.

```ts
    const configValue = new LoggingConfig(
      config.schema.validate({
        appenders: {
          file1: {
            kind: 'file',
            layout: { kind: 'pattern' },
            path: 'path',
          },
        },
      })
    );
```

However, you're right that this could lead to false negatives if the config schema defaults change. If you think that eliminating that risk is more important than the readability here I will change.

Reply:
ok. Let's keep it as is